### ISSUE TITLE
Per-guardrail precision/recall/F1 report the negative class when only one guardrail is present

### ISSUE THREAD

**Issue by WatchTree-19** [alphagov/govuk-chat-evaluation#376]
`AggregateResults._metric_per_guardrail` in `govuk_chat_evaluation/output_guardrails/evaluate.py` assumes sklearn returns one score per guardrail. When `guardrail_names` has exactly one entry it returns one score per *class* instead, and `to_dict` then reports the negative class under the guardrail's name.

## Why

`_expected_actual_guardrails_vectors` builds vectors of width `len(self.guardrail_names)`, and that length is data-dependent (it is the sorted union of keys across the results). With a single guardrail the arrays are shape `(n, 1)`, which sklearn's `type_of_target` classifies as `binary` rather than `multilabel-indicator`. With `average=None` that returns a 2-element array of per-class scores.

`to_dict` then indexes it positionally:

```python
for i, name in enumerate(self.guardrail_names):
    base_metrics[f"Precision [{name}]"] = precisions[i]
```

so `i=0` picks the negative class (the guardrail did *not* fire) and labels it as the guardrail's precision. The real value sits at index 1 and is silently discarded. Nothing raises, and the reported figure is plausible.

## Reproduction

One guardrail, TP=1, FP=1, FN=1, TN=2, so the true precision, recall and F1 are all 0.5.

```python
import numpy as np
from sklearn.metrics import precision_score
from sklearn.utils.multiclass import type_of_target

exp = np.array([[1],[0],[1],[0],[0]])
act = np.array([[1],[1],[0],[0],[0]])

print(type_of_target(exp))                                    # binary
print(precision_score(exp, act, average=None, zero_division=np.nan))
# [0.667 0.5]  -> len 2 for 1 guardrail; index 0 is the negative class
```

```
one guardrail   reported Precision [x] = 0.667   (true value 0.5)
two guardrails  reported Precision [x] = 0.5     (correct)
```

The two-guardrail control is correct, so the fault is specific to the single-column case. sklearn 1.7.2.

The direction matters: the negative class is scored against a TN-heavy base rate, so the guardrail reads as better than it is rather than worse.

## When it fires

Any evaluation run where the data contains exactly one guardrail name, which is the natural shape when tuning a single guardrail in isolation.

`_triggered_metric` is unaffected, since it is 1-D binary and guards the empty case explicitly.

## Possible fix

Pin the label space rather than inferring it, e.g. pass `labels=range(len(self.guardrail_names))`, or assert that the returned array length matches `len(self.guardrail_names)` so the mismatch fails loudly instead of being reported.

Happy to open a PR with a fix and a regression test for the single-guardrail case if that would be useful. I appreciate this repo looks like it is coordinated internally rather than through GitHub, so equally happy for you to just take it.
